tests/unit/s2n_encrypted_extensions_test.c

```diff
@@ -86,41 +87,42 @@ int main(int argc, char **argv)
         const uint8_t EMPTY_SERVER_NAME_EXT_SIZE = 4;
 
         EXPECT_EQUAL(s2n_server_extensions_server_name_send_size(server_conn), EMPTY_SERVER_NAME_EXT_SIZE);
-        EXPECT_EQUAL(s2n_encrypted_extensions_send_size(server_conn), EMPTY_SERVER_NAME_EXT_SIZE);
         EXPECT_SUCCESS(s2n_encrypted_extensions_send(server_conn));
         S2N_STUFFER_LENGTH_WRITTEN_EXPECT_EQUAL(extension_stuffer, EMPTY_SERVER_NAME_EXT_SIZE + ENCRYPTED_EXTENSIONS_HEADER_SIZE);
+
         /* Reset and check */
         server_conn->server_name_used = 0;
-        EXPECT_EQUAL(s2n_encrypted_extensions_send_size(server_conn), 0);
+        EXPECT_SUCCESS(s2n_encrypted_extensions_send(server_conn));
```

Comment:
Is it just my eyes or does it look like you're calling  EXPECT_SUCCESS(s2n_encrypted_extensions_send(server_conn)) twice? Once on line 95 and once on line 90?

Reply:
Nope, that's correct. Once with server_name_used set, and once without it set. We expect the extensions to be sent the first time but not the second.